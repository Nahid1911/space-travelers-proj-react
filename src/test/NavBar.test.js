import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/NavBar';

describe('Navbar Should render as expected', () => {
  it('should work as expected', () => {
    const result = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(result).toMatchSnapshot();
  });
});
