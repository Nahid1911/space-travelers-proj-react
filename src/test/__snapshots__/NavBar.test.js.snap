// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render as expected should work as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbarContainer hstack"
      >
        <div
          class="logoContainer"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt=""
              class="logo mr1"
              src="planet_bigger.png"
            />
          </a>
          <h1>
            Space Travelers' HUB
          </h1>
        </div>
        <div
          class="linksContainer nav"
        >
          <div
            class="nav-item"
          >
            <a
              aria-current="page"
              class="linksStyles mr1 active nav-link active"
              exact="true"
              href="/"
            >
              Rockets 
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="linksStyles mr1 nav-link"
              href="/mission"
            >
              Missions
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="linksStyles borderRight nav-link"
              href="/dragon"
            >
              Dragons
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              class="linksStyles ml1 nav-link"
              href="/myprofile"
            >
              My Profile
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbarContainer hstack"
    >
      <div
        class="logoContainer"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <img
            alt=""
            class="logo mr1"
            src="planet_bigger.png"
          />
        </a>
        <h1>
          Space Travelers' HUB
        </h1>
      </div>
      <div
        class="linksContainer nav"
      >
        <div
          class="nav-item"
        >
          <a
            aria-current="page"
            class="linksStyles mr1 active nav-link active"
            exact="true"
            href="/"
          >
            Rockets 
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="linksStyles mr1 nav-link"
            href="/mission"
          >
            Missions
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="linksStyles borderRight nav-link"
            href="/dragon"
          >
            Dragons
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="linksStyles ml1 nav-link"
            href="/myprofile"
          >
            My Profile
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
